tools/nodejs/src/data_chunk.cpp

```diff
@@ -145,7 +145,20 @@ Napi::Array EncodeDataChunk(Napi::Env env, duckdb::DataChunk &chunk, bool with_t
 				}
 				break;
 			}
-			case duckdb::LogicalTypeId::BLOB:
+			case duckdb::LogicalTypeId::BLOB: {
+				if (with_data) {
+					auto array = Napi::Array::New(env, chunk.size());
+					auto data = duckdb::FlatVector::GetData<duckdb::string_t>(*vec);
+
+					for (size_t i = 0; i < chunk.size(); ++i) {
+						auto value = data[i].GetString();
+						auto buf = Napi::Buffer<char>::Copy(env, value.c_str(), value.length());
```

Comment:
Calling `.GetString()` performs a copy, effectively leading to two copies being performed here. Can we instead use `data[i].GetDataUnsafe()` and `data[i].GetSize()`?

Reply:
Thanks, just pushed a fix!